vllm/v1/worker/gpu_model_runner.py

```diff
@@ -191,6 +192,20 @@ def __init__(
         self.speculative_config = vllm_config.speculative_config
         self.observability_config = vllm_config.observability_config
 
+        reasoning_config = self.vllm_config.reasoning_config
+        has_reasoning_strings = (reasoning_config.think_start_str is not None
+                                 and reasoning_config.think_end_str
+                                 is not None)
+        if has_reasoning_strings:
+            tokenizer = init_tokenizer_from_configs(
+                model_config=self.vllm_config.model_config)
+            reasoning_config.think_start_token_ids = \
+                tokenizer.convert_tokens_to_ids(
+                    tokenizer.tokenize(reasoning_config.think_start_str))
+            reasoning_config.think_end_token_ids = \
+                tokenizer.convert_tokens_to_ids(
+                    tokenizer.tokenize(reasoning_config.think_end_str))
```

Comment:
Let's not do this. Tokenizer shouldn't be initialized inside the GPU runner here.

I prefer to have the think_token_ids computed beforehand.